util/util-internal-validation/src/primitives.ts

```diff
@@ -129,12 +129,16 @@ export const SMALL_QTY: Validator<number, Bytes> = {
  * Hex encoded binary string
  */
 export const BYTES: Validator<Bytes> = {
-    cast(value: unknown): string | ValidationFailure {
-        return this.validate(value) || value as Bytes
+    cast(value: unknown): Bytes | ValidationFailure {
+        if (isBytes(value)) {
+            return value.toLowerCase() as Bytes
+        } else {
+            return new ValidationFailure(value, `{value} is not a hex encoded binary string`)
+        }
     },
     validate(value: unknown): ValidationFailure | undefined {
-        if (isBytes(value)) return
-        return new ValidationFailure(value, `{value} is not a hex encoded binary string`)
+        let i = this.cast(value)
```

Comment:
Here `this.cast()` performs extra `.toLowerCase()` work and potentially creates an extra object. This might be noticeable in a data intensive context.

That's why it is better to do `cast()` like it was originally - `this.validate(value) || (value as Bytes).toLowerCase()`